### ISSUE TITLE
Pipeline could not save when add permission settings

### ISSUE THREAD

**Issue by axot** [spinnaker/spinnaker#3966]
We are using Google group auth for spinnaker, when adding permission to pipeline, save button show errors.
```
spin-orca-54bcbcc84c-qtsht orca 2019-02-13 04:50:15.796 ERROR 1 --- [    handlers-19] c.n.s.orca.q.handler.RunTaskHandler      : [] Error running SavePipelineTask for orchestration[01D3JK1SZ99CNPJGH4DMAGERCQ]
spin-orca-54bcbcc84c-qtsht orca retrofit.RetrofitError: 403
spin-orca-54bcbcc84c-qtsht orca 	at retrofit.RetrofitError.httpError(RetrofitError.java:40)
spin-orca-54bcbcc84c-qtsht orca 	at retrofit.RestAdapter$RestHandler.invokeRequest(RestAdapter.java:388)
spin-orca-54bcbcc84c-qtsht orca 	at retrofit.RestAdapter$RestHandler.invoke(RestAdapter.java:240)
spin-orca-54bcbcc84c-qtsht orca 	at com.sun.proxy.$Proxy86.savePipeline(Unknown Source)
spin-orca-54bcbcc84c-qtsht orca 	at com.netflix.spinnaker.orca.front50.tasks.SavePipelineTask.execute(SavePipelineTask.java:92)
spin-orca-54bcbcc84c-qtsht orca 	at com.netflix.spinnaker.orca.q.handler.RunTaskHandler$handle$1$2$1.invoke(RunTaskHandler.kt:100)
spin-orca-54bcbcc84c-qtsht orca 	at com.netflix.spinnaker.orca.q.handler.RunTaskHandler$handle$1$2$1.invoke(RunTaskHandler.kt:56)
spin-orca-54bcbcc84c-qtsht orca 	at com.netflix.spinnaker.orca.q.handler.RunTaskHandler.withLoggingContext(RunTaskHandler.kt:311)
spin-orca-54bcbcc84c-qtsht orca 	at com.netflix.spinnaker.orca.q.handler.RunTaskHandler.access$withLoggingContext(RunTaskHandler.kt:56)
spin-orca-54bcbcc84c-qtsht orca 	at com.netflix.spinnaker.orca.q.handler.RunTaskHandler$handle$1$2.invoke(RunTaskHandler.kt:99)
spin-orca-54bcbcc84c-qtsht orca 	at com.netflix.spinnaker.orca.q.handler.RunTaskHandler$handle$1$2.invoke(RunTaskHandler.kt:56)
spin-orca-54bcbcc84c-qtsht orca 	at com.netflix.spinnaker.orca.q.handler.AuthenticationAware$sam$java_util_concurrent_Callable$0.call(AuthenticationAware.kt)
spin-orca-54bcbcc84c-qtsht orca 	at com.netflix.spinnaker.security.AuthenticatedRequest.lambda$propagate$0(AuthenticatedRequest.java:97)
spin-orca-54bcbcc84c-qtsht orca 	at com.netflix.spinnaker.orca.q.handler.AuthenticationAware$DefaultImpls.withAuth(AuthenticationAware.kt:49)
spin-orca-54bcbcc84c-qtsht orca 	at com.netflix.spinnaker.orca.q.handler.RunTaskHandler.withAuth(RunTaskHandler.kt:56)
spin-orca-54bcbcc84c-qtsht orca 	at com.netflix.spinnaker.orca.q.handler.RunTaskHandler$handle$1.invoke(RunTaskHandler.kt:98)
spin-orca-54bcbcc84c-qtsht orca 	at com.netflix.spinnaker.orca.q.handler.RunTaskHandler$handle$1.invoke(RunTaskHandler.kt:56)
spin-orca-54bcbcc84c-qtsht orca 	at com.netflix.spinnaker.orca.q.handler.RunTaskHandler$withTask$1.invoke(RunTaskHandler.kt:182)
spin-orca-54bcbcc84c-qtsht orca 	at com.netflix.spinnaker.orca.q.handler.RunTaskHandler$withTask$1.invoke(RunTaskHandler.kt:56)
spin-orca-54bcbcc84c-qtsht orca 	at com.netflix.spinnaker.orca.q.handler.OrcaMessageHandler$withTask$1.invoke(OrcaMessageHandler.kt:49)
spin-orca-54bcbcc84c-qtsht orca 	at com.netflix.spinnaker.orca.q.handler.OrcaMessageHandler$withTask$1.invoke(OrcaMessageHandler.kt:33)
spin-orca-54bcbcc84c-qtsht orca 	at com.netflix.spinnaker.orca.q.handler.OrcaMessageHandler$withStage$1.invoke(OrcaMessageHandler.kt:59)
spin-orca-54bcbcc84c-qtsht orca 	at com.netflix.spinnaker.orca.q.handler.OrcaMessageHandler$withStage$1.invoke(OrcaMessageHandler.kt:33)
spin-orca-54bcbcc84c-qtsht orca 	at com.netflix.spinnaker.orca.q.handler.OrcaMessageHandler$DefaultImpls.withExecution(OrcaMessageHandler.kt:68)
spin-orca-54bcbcc84c-qtsht orca 	at com.netflix.spinnaker.orca.q.handler.RunTaskHandler.withExecution(RunTaskHandler.kt:56)
spin-orca-54bcbcc84c-qtsht orca 	at com.netflix.spinnaker.orca.q.handler.OrcaMessageHandler$DefaultImpls.withStage(OrcaMessageHandler.kt:55)
spin-orca-54bcbcc84c-qtsht orca 	at com.netflix.spinnaker.orca.q.handler.RunTaskHandler.withStage(RunTaskHandler.kt:56)
spin-orca-54bcbcc84c-qtsht orca 	at com.netflix.spinnaker.orca.q.handler.OrcaMessageHandler$DefaultImpls.withTask(OrcaMessageHandler.kt:42)
spin-orca-54bcbcc84c-qtsht orca 	at com.netflix.spinnaker.orca.q.handler.RunTaskHandler.withTask(RunTaskHandler.kt:56)
spin-orca-54bcbcc84c-qtsht orca 	at com.netflix.spinnaker.orca.q.handler.RunTaskHandler.withTask(RunTaskHandler.kt:175)
spin-orca-54bcbcc84c-qtsht orca 	at com.netflix.spinnaker.orca.q.handler.RunTaskHandler.handle(RunTaskHandler.kt:69)
spin-orca-54bcbcc84c-qtsht orca 	at com.netflix.spinnaker.orca.q.handler.RunTaskHandler.handle(RunTaskHandler.kt:56)
spin-orca-54bcbcc84c-qtsht orca 	at com.netflix.spinnaker.q.MessageHandler$DefaultImpls.invoke(MessageHandler.kt:36)
spin-orca-54bcbcc84c-qtsht orca 	at com.netflix.spinnaker.orca.q.handler.OrcaMessageHandler$DefaultImpls.invoke(OrcaMessageHandler.kt)
spin-orca-54bcbcc84c-qtsht orca 	at com.netflix.spinnaker.orca.q.handler.RunTaskHandler.invoke(RunTaskHandler.kt:56)
spin-orca-54bcbcc84c-qtsht orca 	at com.netflix.spinnaker.orca.q.audit.ExecutionTrackingMessageHandlerPostProcessor$ExecutionTrackingMessageHandlerProxy.invoke(ExecutionTrackingMessageHandlerPostProcessor.kt:47)
spin-orca-54bcbcc84c-qtsht orca 	at com.netflix.spinnaker.q.QueueProcessor$pollOnce$1$1.run(QueueProcessor.kt:82)
spin-orca-54bcbcc84c-qtsht orca 	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
spin-orca-54bcbcc84c-qtsht orca 	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
spin-orca-54bcbcc84c-qtsht orca 	at java.lang.Thread.run(Thread.java:748)
```

And there are some google related errors
```
spin-front50-76db55566f-7klfl front50 2019-02-13 04:59:01.731  INFO 1 --- [0.0-8080-exec-6] c.n.s.front50.model.GcsStorageService    : Wrote serviceAccounts 'ee0b2426-2c20-4678-b2b6-cb70e0936ad5@managed-service-account'
spin-front50-76db55566f-7klfl front50 2019-02-13 04:59:01.766  INFO 1 --- [0.0-8080-exec-6] c.n.spinnaker.fiat.shared.FiatService    : ---> HTTP PUT http://spin-fiat.spinnaker:7003/roles/ee0b2426-2c20-4678-b2b6-cb70e0936ad5%40managed-service-account
spin-fiat-84f4467b96-nh7r2 fiat Caused by: com.google.api.client.googleapis.json.GoogleJsonResponseException: 400 Bad Request
spin-fiat-84f4467b96-nh7r2 fiat {
spin-fiat-84f4467b96-nh7r2 fiat   "code" : 400,
spin-fiat-84f4467b96-nh7r2 fiat   "errors" : [ {
spin-fiat-84f4467b96-nh7r2 fiat     "domain" : "global",
spin-fiat-84f4467b96-nh7r2 fiat     "message" : "Invalid Input: memberKey",
spin-fiat-84f4467b96-nh7r2 fiat     "reason" : "invalid"
spin-fiat-84f4467b96-nh7r2 fiat   } ],
spin-fiat-84f4467b96-nh7r2 fiat   "message" : "Invalid Input: memberKey"
spin-fiat-84f4467b96-nh7r2 fiat }
spin-fiat-84f4467b96-nh7r2 fiat 	at com.google.api.client.googleapis.json.GoogleJsonResponseException.from(GoogleJsonResponseException.java:150)
spin-fiat-84f4467b96-nh7r2 fiat 	at com.google.api.client.googleapis.services.json.AbstractGoogleJsonClientRequest.newExceptionOnError(AbstractGoogleJsonClientRequest.java:113)
spin-fiat-84f4467b96-nh7r2 fiat 	at com.google.api.client.googleapis.services.json.AbstractGoogleJsonClientRequest.newExceptionOnError(AbstractGoogleJsonClientRequest.java:40)
spin-fiat-84f4467b96-nh7r2 fiat 	at com.google.api.client.googleapis.services.AbstractGoogleClientRequest$1.interceptResponse(AbstractGoogleClientRequest.java:417)
spin-fiat-84f4467b96-nh7r2 fiat 	at com.google.api.client.http.HttpRequest.execute(HttpRequest.java:1132)
spin-fiat-84f4467b96-nh7r2 fiat 	at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.executeUnparsed(AbstractGoogleClientRequest.java:515)
spin-fiat-84f4467b96-nh7r2 fiat 	at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.executeUnparsed(AbstractGoogleClientRequest.java:448)
spin-fiat-84f4467b96-nh7r2 fiat 	at com.google.api.client.googleapis.services.AbstractGoogleClientRequest.execute(AbstractGoogleClientRequest.java:565)
spin-fiat-84f4467b96-nh7r2 fiat 	at com.netflix.spinnaker.fiat.roles.google.GoogleDirectoryUserRolesProvider.loadRoles(GoogleDirectoryUserRolesProvider.java:162)
spin-fiat-84f4467b96-nh7r2 fiat 	... 101 common frames omitted
```

and front50
```
spin-front50-76db55566f-7klfl front50 2019-02-13 04:50:15.740  WARN 1 --- [0.0-8080-exec-2] c.n.s.f.c.ServiceAccountsController      : Error syncing users
spin-front50-76db55566f-7klfl front50
spin-front50-76db55566f-7klfl front50 retrofit.RetrofitError: 500
spin-front50-76db55566f-7klfl front50 	at retrofit.RetrofitError.httpError(RetrofitError.java:40) ~[retrofit-1.9.0.jar:na]
spin-front50-76db55566f-7klfl front50 	at retrofit.RestAdapter$RestHandler.invokeRequest(RestAdapter.java:388) ~[retrofit-1.9.0.jar:na]
spin-front50-76db55566f-7klfl front50 	at retrofit.RestAdapter$RestHandler.invoke(RestAdapter.java:240) ~[retrofit-1.9.0.jar:na]
spin-front50-76db55566f-7klfl front50 	at com.sun.proxy.$Proxy107.loginWithRoles(Unknown Source) ~[na:na]
spin-front50-76db55566f-7klfl front50 	at com.netflix.spinnaker.fiat.shared.FiatService$loginWithRoles.call(Unknown Source) ~[na:na]
spin-front50-76db55566f-7klfl front50 	at com.netflix.spinnaker.front50.controllers.ServiceAccountsController.syncUsers(ServiceAccountsController.groovy:78) [front50-web.jar:na]
spin-front50-76db55566f-7klfl front50 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.8.0_191]
spin-front50-76db55566f-7klfl front50 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[na:1.8.0_191]
spin-front50-76db55566f-7klfl front50 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.8.0_191]
spin-front50-76db55566f-7klfl front50 	at java.lang.reflect.Method.invoke(Method.java:498) ~[na:1.8.0_191]
spin-front50-76db55566f-7klfl front50 	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite$PogoCachedMethodSiteNoUnwrapNoCoerce.invoke(PogoMetaMethodSite.java:210) [groovy-all-2.4.15.jar:2.4.15]
spin-front50-76db55566f-7klfl front50 	at org.codehaus.groovy.runtime.callsite.PogoMetaMethodSite.callCurrent(PogoMetaMethodSite.java:59) [groovy-all-2.4.15.jar:2.4.15]
spin-front50-76db55566f-7klfl front50 	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.callCurrent(AbstractCallSite.java:169) [groovy-all-2.4.15.jar:2.4.15]
spin-front50-76db55566f-7klfl front50 	at com.netflix.spinnaker.front50.controllers.ServiceAccountsController.createServiceAccount(ServiceAccountsController.groovy:56) [front50-web.jar:na]
spin-front50-76db55566f-7klfl front50 	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:1.8.0_191]
spin-front50-76db55566f-7klfl front50 	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[na:1.8.0_191]
spin-front50-76db55566f-7klfl front50 	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:1.8.0_191]
spin-front50-76db55566f-7klfl front50 	at java.lang.reflect.Method.invoke(Method.java:498) ~[na:1.8.0_191]
```

**Comment by axot** [comment]
Hi @ajordens , any advice? Thank you.

**Comment by axot** [comment]
I think the email of `ee0b2426-2c20-4678-b2b6-cb70e0936ad5%40managed-service-account` should be google email, is it right?

**Comment by dibyom** [comment]
Hi @axot what version of spinnaker are you using?

**Comment by axot** [comment]
Hello @dibyom , we are using 1.12.2 of spinnaker

**Comment by dibyom** [comment]
thanks! I can reproduce this locally and will work on a fix.

**Comment by dibyom** [comment]
Fix was cherry-picked on to release-1.12 and should be in 1.12.3
